### PR TITLE
feat: Add StyleProp types

### PR DESCRIPTION
**Pull request by alanbsmith**
## Summary

Fixes: #865

## Release Category
Types

---

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [standard MDX template](https://github.com/Workday/canvas-kit/discussions/1131)

## For the Reviewer

- [x] PR title is short and descriptive
- [x] PR summary describes the change (Fixes/Resolves linked correctly)
- [x] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [x] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Where Should the Reviewer Start?

`modules/react/layout/index.ts`

## Areas for Feedback? (optional)

- [x] Code
- [ ] Documentation
- [ ] Testing
- [ ] Codemods

## Testing Manually

n/a

## Thank You Gif (optional)

a smiling cartoon corgi

### CODE REVIEW

File: modules/react/layout/lib/Box.tsx [hunk 1]
```diff
@@ -4,31 +4,22 @@ import isPropValid from '@emotion/is-prop-valid';
...
-  FlexItemStyleProps &
-  GridItemStyleProps &
-  LayoutStyleProps &
-  OtherStyleProps &
-  PositionStyleProps &
-  SpaceStyleProps &
-  TextStyleProps & {
-    children?: React.ReactNode;
-  };
+export type BoxProps = CommonStyleProps & {
```
Comment (mannycarrera4) [inline modules/react/layout/lib/Box.tsx:20]: this is an exported type, do we want JS docs here?

Comment (mannycarrera4) [inline modules/react/layout/lib/Box.tsx:20]: add js docs

**Review by mannycarrera4 — APPROVED** [review]
gracias!

File: modules/react/layout/lib/Box.tsx (hunk 1 above)
Comment (alanbsmith) [inline modules/react/layout/lib/Box.tsx:20]:
```suggestion
/**
 * Box Props
 * ---
 * Common style props + children
 * 
 * - background
 * - border
 * - color
 * - depth
 * - flexItem
 * - gridItem
 * - layout
 * - other
 * - position
 * - space
 * - text
 */
```

File: modules/react/layout/lib/Box.tsx [hunk 2]
```diff
@@ -4,31 +4,22 @@ import isPropValid from '@emotion/is-prop-valid';
...
+import {color} from './utils/color';
+import {depth} from './utils/depth';
+import {flexItem} from './utils/flexItem';
+import {gridItem} from './utils/gridItem';
+import {layout} from './utils/layout';
+import {other} from './utils/other';
+import {position} from './utils/position';
+import {space} from './utils/space';
+import {text} from './utils/text';
+import {CommonStyleProps} from './utils/styleProps';
```
Comment (alanbsmith) [inline modules/react/layout/lib/Box.tsx:19]:
```suggestion

/**
 * Box Props
 * ---
 * Common style props + children
 * 
 * - background
 * - border
 * - color
 * - depth
 * - flexItem
 * - gridItem
 * - layout
 * - other
 * - position
 * - space
 * - text
 */
```

**Pull request merged**